R/article.R

```diff
@@ -281,19 +281,67 @@ mnras_article <- function(..., keep_tex = TRUE, fig_caption = TRUE) {
   )
 }
 
+#' @param document_style For \code{oup_article}, one of "contemporary", "modern", or "traditional" setting overall style of document. Defaults to "contemporary".
+#' @param papersize For \code{oup_article}, one of "large", "medium", or "small" setting output page size. Defaults to "large".
+#' @param namedate For \code{oup_article}, a logical variable indicating if natbib citations should be in name-date format. Defaults to \code{FALSE}.
+#' @param onecolumn For \code{oup_article}, a logical variable indicating if one column formatting should be used. Defaults to \code{FALSE}.
 #' @section \code{oup_article}: Format for creating submissions to many Oxford University Press
 #'   journals. Adapted from
 #'   \url{https://academic.oup.com/journals/pages/authors/preparing_your_manuscript}
-#'   and \url{https://academic.oup.com/icesjms/pages/General_Instructions}.
+#'   and the \code{oup-authoring-template} available on CTAN at
+#'   \url{https://www.ctan.org/pkg/oup-authoring-template}.
 #' @export
 #' @rdname article
 oup_article <- function(
   ..., keep_tex = TRUE,
-  md_extensions = c("-autolink_bare_uris")
+  md_extensions = c("-autolink_bare_uris"),
+  journal=NULL,pandoc_args=NULL,
+  number_sections=TRUE,
+  citation_package=c("natbib","default"),
+  papersize=c("large","medium","small"),
+  document_style=c("contemporary","modern","traditional"),
+  namedate=FALSE,onecolumn=FALSE
 ) {
+  citation_package <- match.arg(citation_package)
+  if (length(citation_package) > 1) stop("Please choose just one ", dQuote("citation_package"))
+  papersize <- match.arg(papersize)
+  if (length(papersize) > 1) stop("Please choose just one ", dQuote("papersize"))
+  document_style <- match.arg(document_style)
+  if (length(document_style) > 1) stop("Please choose just one ", dQuote("document_style"))
+
+  args <- c(
+    journal = journal,
+    papersize = papersize,
+    document_style = document_style
+  )
+
+  # Convert to pandoc arguments
+  pandoc_arg_list <- mapply(rmarkdown::pandoc_variable_arg, names(args), args,
+                            SIMPLIFY = FALSE, USE.NAMES = FALSE)
+
+  # namedate
+  if (namedate)
+    pandoc_arg_list = c(pandoc_arg_list,
+                        rmarkdown::pandoc_variable_arg("namedate"))
+
+  # onecolumn
+  if (onecolumn)
+    pandoc_arg_list = c(pandoc_arg_list,
+                        rmarkdown::pandoc_variable_arg("onecolumn"))
+
+  # section numbering
+  if (!number_sections)
+    pandoc_arg_list = c(pandoc_arg_list,
+                        rmarkdown::pandoc_variable_arg("unnumsec"))
+
+  pandoc_arg_list <- unlist(pandoc_arg_list)
+
   pdf_document_format(
-    "oup",
-    keep_tex = keep_tex, md_extensions = md_extensions, ...
+    "oup", keep_tex = keep_tex,
+    md_extensions=md_extensions,
+    citation_package = citation_package,
+    pandoc_args = c(pandoc_arg_list, pandoc_args),number_sections=FALSE,
```

Comment:
The thing is that the OUP cls file has an entirely different mechanism for numbering sections via the `unnumsec` option to the document class. So as to be able to use the standard `number_sections` YAML entry, but not have double and potentially conflicting code in the output, I thought I needed to set a custom `unnumsec` argument to activate this mechanism, but then unset the standard mechanism via the `number_sections` argument to `pdf_document`. But this was because I never went over the default latex pandoc template. I now see where in that document how section numbering comes about and will adapt my template appropriately to take advantage of the existing mechanism.

Reply:
I don't mind setting the `number_section = FALSE`. However, if we do so we should not expose the argument in the format function. Just hard code the FALSE to prevent user from changing.

Reply:
I already pushed something that fixes this issue and completely eliminates the confusion. I think this is the best solution.